src/documents/matching.py

```diff
@@ -60,14 +60,15 @@ def match_tags(document, classifier):
 def matches(matching_model, document):
     search_kwargs = {}
 
-    document_content = document.content.lower()
-
     # Check that match is not empty
     if matching_model.match.strip() == "":
         return False
 
     if matching_model.is_insensitive:
         search_kwargs = {"flags": re.IGNORECASE}
+        document_content = document.content.lower()
```

Comment:
With the `re.IGNORECASE`, there's actually no reason at all to lower case the contents.

`document_content = document.content` would be enough in that case

Reply:
makes sense.